TablePagination className not merged

<!-- Provide a general summary of the issue in the Title above -->
The TablePagination component does not use the clsx function to merge the classes.root and the className property. 

<!-- Checked checkbox should look like this: [x] -->

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯
This leads to classes.root being discarded and replaced by className when both are supplied.

## Expected Behavior 🤔
The clsx application should probably be used like in other components like TableBody, TableCell, etc.

## Steps to Reproduce 🕹
In the following example, you may look at the console html and see that the classes are missing on the MuiTablePagination-root div.
[https://codesandbox.io/s/create-react-app-8ho0q](https://codesandbox.io/s/create-react-app-8ho0q)

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Material-UI | v4.4.0  |
| React       | 16.9.0  |
| Chrome     |  76.0.3809       |

@oliviertassinari It is strange because if I add your test and run `yarn test` in the root folder I do not see the expected error, instead I see:
```
error Command failed with exit code 5.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 5.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
However, if I go to the material-ui package and run the `yarn test` from there I get the expected error:
```
2212 passing (13s)
      AssertionError: expected false to equal true
      + expected - actual

      -false
      +true

      at Context.strictEqual (packages\material-ui\src\test-utils/describeConformance.js:147:12)
      at processImmediate (internal/timers.js:439:21)
```
Any idea why I get the unexpected error with the standard `yarn test` from the root folder?

--- a/packages/material-ui-lab/src/SpeedDialIcon/SpeedDialIcon.js
+++ b/packages/material-ui-lab/src/SpeedDialIcon/SpeedDialIcon.js
@@ -40,7 +40,7 @@ export const styles = theme => ({
 });
 
 const SpeedDialIcon = React.forwardRef(function SpeedDialIcon(props, ref) {
-  const { classes, icon: iconProp, open, openIcon: openIconProp, ...other } = props;
+  const { className, classes, icon: iconProp, open, openIcon: openIconProp, ...other } = props;
 
   const iconClassName = clsx(classes.icon, {
     [classes.iconOpen]: open,
@@ -49,16 +49,16 @@ const SpeedDialIcon = React.forwardRef(function SpeedDialIcon(props, ref) {
 
   const openIconClassName = clsx(classes.openIcon, { [classes.openIconOpen]: open });
 
-  function formatIcon(icon, className) {
+  function formatIcon(icon, newClassName) {
     if (React.isValidElement(icon)) {
-      return React.cloneElement(icon, { className });
+      return React.cloneElement(icon, { className: newClassName });
     }
 
     return icon;
   }
 
   return (
-    <span className={classes.root} ref={ref} {...other}>
+    <span className={clsx(classes.root, className)} ref={ref} {...other}>
       {openIconProp ? formatIcon(openIconProp, openIconClassName) : null}
       {iconProp ? formatIcon(iconProp, iconClassName) : <AddIcon className={iconClassName} />}
     </span>
@@ -75,6 +75,10 @@ SpeedDialIcon.propTypes = {
    * See [CSS API](#css) below for more details.
    */
   classes: PropTypes.object,
+  /**
+   * @ignore
+   */
+  className: PropTypes.string,
   /**
    * The icon to display in the SpeedDial Floating Action Button.
    */

--- a/packages/material-ui/src/TablePagination/TablePagination.js
+++ b/packages/material-ui/src/TablePagination/TablePagination.js
@@ -78,6 +78,7 @@ const TablePagination = React.forwardRef(function TablePagination(props, ref) {
     ActionsComponent = TablePaginationActions,
     backIconButtonProps,
     classes,
+    className,
     colSpan: colSpanProp,
     component: Component = TableCell,
     count,
@@ -102,7 +103,7 @@ const TablePagination = React.forwardRef(function TablePagination(props, ref) {
   const MenuItemComponent = SelectProps.native ? 'option' : MenuItem;
 
   return (
-    <Component className={classes.root} colSpan={colSpan} ref={ref} {...other}>
+    <Component className={clsx(classes.root, className)} colSpan={colSpan} ref={ref} {...other}>
       <Toolbar className={classes.toolbar}>
         <div className={classes.spacer} />
         {rowsPerPageOptions.length > 1 && (
@@ -169,6 +170,10 @@ TablePagination.propTypes = {
    * See [CSS API](#css) below for more details.
    */
   classes: PropTypes.object.isRequired,
+  /**
+   * @ignore
+   */
+  className: PropTypes.string,
   /**
    * @ignore
    */

--- a/packages/material-ui/src/test-utils/describeConformance.js
+++ b/packages/material-ui/src/test-utils/describeConformance.js
@@ -137,13 +137,15 @@ function testRootClass(element, getOptions) {
       return;
     }
 
-    const wrapper = mount(element);
+    const className = randomStringValue();
+    const wrapper = mount(React.cloneElement(element, { className }));
 
     // we established that the root component renders the outermost host previously. We immediately
     // jump to the host component because some components pass the `root` class
     // to the `classes` prop of the root component.
     // https://github.com/mui-org/material-ui/blob/f9896bcd129a1209153106296b3d2487547ba205/packages/material-ui/src/OutlinedInput/OutlinedInput.js#L101
     assert.strictEqual(findOutermostIntrinsic(wrapper).hasClass(classes.root), true);
+    assert.strictEqual(findOutermostIntrinsic(wrapper).hasClass(className), true);
   });
 }
 

